### PR TITLE
Skip migration if user doesn't exist

### LINKED ISSUE #4455: [10.7 unstable] Failure during the migration of DisplayPreferences database
**System (please complete the following information):**
 - OS: Docker
 - Virtualization: [e.g. Docker, KVM, LXC]
 - Clients: [Browser, Android, Fire Stick, etc.]
 - Browser: [e.g. Firefox 72, Chrome 80, Safari 13]
 - Jellyfin Version: 20201110.2-unstable 10.7
 - Playback: [Direct Play, Remux, Direct Stream, Transcode] 
 - Installed Plugins: [e.g. none, Fanart, Anime, etc.]
 - Reverse Proxy: [e.g. none, nginx, apache, etc.]
 - Base URL: [e.g. none, yes: /example]
 - Networking: [e.g. Host, Bridge/NAT]
 - Storage: [e.g. local, NFS, cloud]

**To Reproduce**
1. Upgrade from stable to unstable

**Logs**
```
[2020-11-10 02:11:10.730 +00:00] [INF] [1] Jellyfin.Server.Migrations.MigrationRunner: Applying migration '"MigrateDisplayPreferencesDatabase"'
[2020-11-10 02:11:11.883 +00:00] [ERR] [1] Jellyfin.Server.Migrations.MigrationRunner: Could not apply migration '"MigrateDisplayPreferencesDatabase"'
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.Internal.NoopExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Jellyfin.Server.Implementations.JellyfinDb.SaveChanges()
   at Jellyfin.Server.Migrations.Routines.MigrateDisplayPreferencesDb.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.Run(CoreAppHost host, ILoggerFactory loggerFactory)
   at Jellyfin.Server.Program.StartApp(StartupOptions options)
[2020-11-10 02:11:11.885 +00:00] [FTL] [1] Main: Error while starting server.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.Internal.NoopExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Jellyfin.Server.Implementations.JellyfinDb.SaveChanges()
   at Jellyfin.Server.Migrations.Routines.MigrateDisplayPreferencesDb.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.Run(CoreAppHost host, ILoggerFactory loggerFactory)
   at Jellyfin.Server.Program.StartApp(StartupOptions options)
[2020-11-10 02:11:11.890 +00:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "CoreAppHost"
[2020-11-10 02:11:11.898 +00:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "EmbyTV"
[2020-11-10 02:11:11.903 +00:00] [INF] [2] Main: Received a SIGTERM signal, shutting down
```

### PR DESCRIPTION
**Pull request by crobibero**
Fixes #4455

### CODE REVIEW

**Review by cvium — APPROVED** [review]

**Review by BaronGreenback — APPROVED** [review]

**Comment by crobibero** [comment]
On hold for further testing

**Comment by crobibero** [comment]
Confirmed that this fixes the reported issue

File: Jellyfin.Server/Migrations/Routines/MigrateDisplayPreferencesDb.cs [hunk 1]
```diff
@@ -86,11 +94,19 @@ public void Perform()
                         continue;
                     }
 
+                    var dtoUserId = new Guid(result[1].ToBlob());
+                    var existingUser = _userManager.GetUserById(dtoUserId);
+                    if (existingUser == null)
+                    {
+                        _logger.LogWarning("User with ID {userId} does not exist in the database, skipping migration.", dtoUserId);
```
Comment (cvium) [inline Jellyfin.Server/Migrations/Routines/MigrateDisplayPreferencesDb.cs:101]: Is it Pascal or Camel case for log formatting?

Comment (crobibero) [inline Jellyfin.Server/Migrations/Routines/MigrateDisplayPreferencesDb.cs:101]:
```suggestion
                        _logger.LogWarning("User with ID {UserId} does not exist in the database, skipping migration.", dtoUserId);
```

**Pull request merged**